core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -824,4 +824,49 @@ class PlaintextConsumerTest extends BaseConsumerTest {
 
     assertThrows(classOf[WakeupException], () => consumer.position(topicPartition, Duration.ofSeconds(100)))
   }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
+  def testCloseLeavesGroupOnInterrupt(quorum: String, groupProtocol: String): Unit = {
```

Comment:
should we add a very similar test for close(0)?

Reply:
I worked up the test, but I ran it 10 times and it failed twice. The 'leave group on 0 timeout' _is_ a best effort, so the timing may not work out that we can send off the request before the `ConsumerNetworkThread` shuts down.

Reply:
ack, makes sense. It's still important to ensure we run callbacks on close(0) but that is covered on `testCloseLeavesGroupDespiteOnPartitionsLostError`, all good.